### PR TITLE
feat: migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by anshumanv**

### CODE REVIEW

**Comment by alexander-akait** [comment]
We need improve our `scripts` on test we don't need link, also let's fix lint problems

**Comment by anshumanv** [comment]
some npm err 😞

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  build:
+    name: Tests and Coverage - ${{ matrix.os }} - Node v${{ matrix.node-version }}, Webpack ${{ matrix.webpack-version }}
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [10.x, 12.x, 14.x]
+        webpack-version: [4, latest]
```
Comment (snitin315) [inline .github/workflows/nodejs.yml:50]: I think not yet compatible with webpack-5.

Comment (anshumanv) [reply .github/workflows/nodejs.yml:50]: oh yep lets avoid it

**Comment by alexander-akait** [comment]
Let's rebase

**Comment by anshumanv** [comment]
@alexander-akait some problem with prettier 😞

**Comment by alexander-akait** [comment]
We don't need run `lint` on `tests` and please fix lint problems too

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "changelog": "npm run changelog:generate && npm run changelog:add",
     "changelog:add": "git add CHANGELOG.md",
     "changelog:generate": "github_changelog_generator --future-release $npm_package_version",
-    "coverage": "npm test && nyc report --reporter=lcov > coverage/coverage.lcov && codecov",
```
Comment (alexander-akait) [inline package.json:16]: Wrong fix, we should not run link on test

Comment (anshumanv) [reply package.json:16]: just `jest --coverage`?

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
...
+    "coverage": "nyc jest",
```
Comment (snitin315) [inline package.json:16]: We are not using `jest` here `mocha` instead

Comment (anshumanv) [reply package.json]: oh yep, maybe we should migrate..

File: package.json (hunk 2 above)
Comment (anshumanv) [reply package.json:16]: done

File: test/installer.test.js [hunk 4]
```diff
@@ -427,12 +430,13 @@ describe('installer', () => {
...
-              expect(this.sync.calls[1].arguments[1]).toEqual([
-                'install',
-                'react',
-                '--save',
-              ]);
+              // expect(this.sync.calls[1].arguments[1]).toEqual([
+              //   'install',
+              //   'react',
+              //   '--save',
+              // ]);
```
Comment (alexander-akait) [inline test/installer.test.js:439]: It was work before, why we comment it?

Comment (anshumanv) [reply test/installer.test.js:439]: diff in test due to peer dep 
image

Comment (anshumanv) [reply test/installer.test.js:439]: peer dep warning, should we update the test with this? I don't think it's correct, we should update the test later

Comment (alexander-akait) [reply test/installer.test.js:439]: Yep, please update

Comment (anshumanv) [reply test/installer.test.js]: done

File: .github/workflows/nodejs.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  build:
+    name: Tests and Coverage - ${{ matrix.os }} - Node v${{ matrix.node-version }}, Webpack ${{ matrix.webpack-version }}
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [10.x, 12.x, 14.x]
```
Comment (snitin315) [inline .github/workflows/nodejs.yml:49]:
```suggestion
        node-version: [12.x, 14.x]
```

File: package.json [hunk 6]
```diff
@@ -13,7 +13,7 @@
     "changelog": "npm run changelog:generate && npm run changelog:add",
     "changelog:add": "git add CHANGELOG.md",
     "changelog:generate": "github_changelog_generator --future-release $npm_package_version",
-    "coverage": "npm test && nyc report --reporter=lcov > coverage/coverage.lcov && codecov",
+    "coverage": "nyc --no-clean mocha",
```
Comment (alexander-akait) [inline package.json:16]: Does it real run tests?

Comment (anshumanv) [reply package.json:16]: yep yep why?

Comment (anshumanv) [reply package.json:16]: you can check log

File: .github/workflows/nodejs.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Using Node v${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Install webpack ${{ matrix.webpack-version }}
+        run: yarn add -W webpack@${{ matrix.webpack-version }}
```
Comment (snitin315) [inline .github/workflows/nodejs.yml:66]: This step is not needed for now + we have npm here. I think it is safe to remove.

File: .github/workflows/nodejs.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+      - name: Run tests and generate coverage
+        run: |
+          npm run test
+          npm run coverage
```
Comment (snitin315) [inline .github/workflows/nodejs.yml:71]: In logs, it seems we are running tests twice?

**Review by alexander-akait — APPROVED** [review]
We need migrate on jest

**Comment by anshumanv** [comment]
Will fix coverage and jest next

**Pull request merged**